### PR TITLE
[Feature] Add support for Identity Column in `databricks_sql_table`

### PR DESCRIPTION
**Pull request by hectorcast-db**
## Changes
Add support for Identity Column in `databricks_sql_table`

## Tests

- [X] `make test` run locally
- [X] relevant change in `docs/` folder
- [X] covered with integration tests in `internal/acceptance`
- [X] relevant acceptance tests are passing
- [X] using Go SDK

### CODE REVIEW

File: catalog/resource_sql_table.go [hunk 1]
```diff
@@ -108,6 +120,28 @@ func parseComment(s string) string {
 	return strings.ReplaceAll(strings.ReplaceAll(s, `\'`, `'`), `'`, `\'`)
 }
 
+func reconstrutIdentity(c *SqlColumnInfo) (IdentityColumn, error) {
```
Comment (mgyucht) [inline catalog/resource_sql_table.go:123]:
```suggestion
func reconstructIdentity(c *SqlColumnInfo) (IdentityColumn, error) {
```

File: catalog/resource_sql_table.go [hunk 2]
```diff
@@ -22,12 +23,23 @@ import (
...
 }
 
+type TypeJson struct {
+	Metadata map[string]any `json:"metadata,omitempty"`
+}
+
+type IdentityColumn string
+
+const IdentityColumnAlways IdentityColumn = "always"
+const IdentityColumnDefault IdentityColumn = "default"
```
Comment (mgyucht) [inline catalog/resource_sql_table.go:41]: Maybe for clarity we can have a constant to indicate no identity column?
```suggestion
const IdentityColumnDefault IdentityColumn = "default"
const IdentityColumnNone IdentityColumn = ""
```

File: catalog/resource_sql_table.go [hunk 3]
```diff
@@ -171,7 +205,22 @@ func (ti *SqlTableInfo) getOrCreateCluster(clusterName string, clustersAPI clust
 	return aclCluster.ClusterID, nil
 }
 
+func (ci *SqlColumnInfo) getColumnType() string {
+	var colType string
+	switch ci.Identity {
+	case IdentityColumnAlways:
+		colType = "BIGINT GENERATED ALWAYS AS IDENTITY"
```
Comment (mgyucht) [inline catalog/resource_sql_table.go:212]: Is the colType always BIGINT when using an identity column? What about if a user specified a different type?

File: docs/resources/sql_table.md [hunk 4]
```diff
@@ -137,6 +177,8 @@ Currently, changing the column definitions for a table will require dropping and
 
 * `name` - User-visible name of column
 * `type` - Column type spec (with metadata) as SQL text. Not supported for `VIEW` table_type.
+* `identity` - (Optional) Whether field is an identity column. Can be `default`, `always` or unset.
+It is unset by default.
```
Comment (mgyucht) [inline docs/resources/sql_table.md:181]:
```suggestion
* `identity` - (Optional) Whether field is an identity column. Can be `default`, `always` or unset. It is unset by default.
```

File: internal/acceptance/sql_table_test.go [hunk 5]
```diff
@@ -72,6 +72,70 @@ func TestUcAccResourceSqlTable_Managed(t *testing.T) {
 	})
 }
 
+func TestUcAccResourceSqlTableWithIdentityColumn_Managed(t *testing.T) {
+	if os.Getenv("GOOGLE_CREDENTIALS") != "" {
+		skipf(t)("databricks_sql_table resource not available on GCP")
+	}
+	unityWorkspaceLevel(t, step{
```
Comment (mgyucht) [inline internal/acceptance/sql_table_test.go:79]: Make sure to merge latest main. These are exported there.

File: internal/acceptance/sql_table_test.go [hunk 6]
```diff
@@ -72,6 +72,70 @@ func TestUcAccResourceSqlTable_Managed(t *testing.T) {
...
+			table_type         = "MANAGED"
+			properties         = {
+				that      = "this"
+				something = "else2"
+			}
+			
+			column {
+				name      = "id"
+				type      = "bigint"
+				identity  = "default"
```
Comment (mgyucht) [inline internal/acceptance/sql_table_test.go:128]: Can we try making this `always`? Just to test changing the identity column, if that is allowed.

**Review by mgyucht — APPROVED** [review]
Looking good, a few small questions but otherwise seems good to me.

File: internal/acceptance/sql_table_test.go (hunk 6 above)
Comment (hectorcast-db) [reply internal/acceptance/sql_table_test.go:128]: Not allowed. Added validation.

**Pull request merged**